cocotb/regression.py

```diff
@@ -201,13 +204,19 @@ def _discover_tests() -> Iterable[Test]:
 
                     yield test
 
-                # only look in first module for all functions and don't complain if all functions are not found
-                break
+                # Clear not_found_tests for the next module search
+                tests = not_found_tests.copy()
```

Comment:
```suggestion
                # Use the non-matching test names in the next module search
                tests = not_found_tests
```

Reply:
Right, I forgot to delete it.

Reply:
Changed in 368e81fd804297276d9e9ac6a7138b256d7f1b63.